### PR TITLE
feat: adding url schema loader functionality

### PR DESCRIPTION
**Pull request by maticardenas**
Adding functionality for loading a schema from a `url` path.

This can be useful for situations where you need to load local tests and you hold schemas in a separate repository / server.

- Adding `UrlStaticSchemaLoader` to handle the cast of the schema content.
- Checking if `schema_file_path` is an `url` and assigning corresponding loader accordingly.
- Adding tests.

This is related to issue #282

### CODE REVIEW

File: openapi_tester/utils.py [hunk 1]
```diff
@@ -58,3 +59,17 @@ def lazy_combinations(options_list: Sequence[dict[str, Any]]) -> Iterator[dict]:
     for i in range(2, len(options_list) + 1):
         for combination in combinations(options_list, i):
             yield merge_objects(combination)
+
+
+def is_path_an_url(path: str) -> bool:
```
Comment (sondrelg) [inline openapi_tester/utils.py:64]: Think we could use the django url validator directly: https://github.com/django/django/blob/stable/1.3.x/django/core/validators.py#L45

File: openapi_tester/loaders.py [hunk 2]
```diff
@@ -258,3 +259,28 @@ def load_schema(self) -> dict[str, Any]:
...
+        self.url = url
+
+    def load_schema(self) -> dict[str, Any]:
+        """
+        Loads a static OpenAPI schema from url, and parses it to a python dict.
+
+        :return: Schema contents as a dict
+        :raises: ImproperlyConfigured
+        """
+        response = httpx.get(self.url)
```
Comment (sondrelg) [inline openapi_tester/loaders.py:280]: Better to use a builtin library like requests

**Review by sondrelg — CHANGES_REQUESTED** [review]
Looks good, but I'd rather not pull in httpx as a dependency for this. Better do use a builtin library like urllib or requests. Otherwise we're forcing users to potentially pull in an extra dependency for no real gain.

File: openapi_tester/utils.py (hunk 1 above)
Comment (maticardenas) [reply openapi_tester/utils.py]: Nice, I hadn't considered it. Changed it to use django's validator.

File: openapi_tester/loaders.py (hunk 2 above)
Comment (maticardenas) [reply openapi_tester/loaders.py]: Replaced `httpx` with `requests`

**Comment by sondrelg** [comment]
Looks great @maticardenas 👏

**Pull request merged**